peachjam/views/causelist.py

```diff
@@ -40,8 +41,12 @@ class CauseListListView(TemplateView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-
-        context["court_classes"] = CourtClass.objects.prefetch_related("courts")
+        # get only court classes and courts that have cause lists, and cache the result
+        court_classes_with_cause_lists = cache.get_or_set(
+            "court_classes_with_causelists",
+            CourtClass.get_court_classes_with_cause_lists(),
+        )
```

Comment:
do we need a timeout?

Reply:
yes

Reply:
set a default timeout (30  mins?) on the default cache in settings.py